File: src/main/java/kr/co/lunatalk/domain/exhibition/dto/response/ExhibitionFindOneResponse.java
package kr.co.lunatalk.domain.exhibition.dto.response;

import io.swagger.v3.oas.annotations.media.Schema;
import kr.co.lunatalk.domain.exhibition.domain.Exhibition;
import kr.co.lunatalk.domain.exhibition.domain.ExhibitionVisibility;
import kr.co.lunatalk.domain.exhibition.dto.ExhibitionProductDto;
import kr.co.lunatalk.domain.product.domain.Product;
import kr.co.lunatalk.domain.product.dto.FindProductDto;

import java.time.LocalDateTime;
import java.util.List;

public record ExhibitionFindOneResponse(
	@Schema(description = "기획전 ID")
	Long exhibitionId,
	@Schema(description = "기획전 이름")
	String title,
	@Schema(description = "기획전 설명")
	String description,

	@Schema(description = "기획전 노출 여부")
	ExhibitionVisibility visibility,

	@Schema(description = "기획전 시작일")
	LocalDateTime startAt,

	@Schema(description = "기획전 종료일")
	LocalDateTime endAt,

	@Schema(description = "연동된 상품들")
	List<ExhibitionProductDto> products
) {

	public static ExhibitionFindOneResponse from(Exhibition exhibition, List<ExhibitionProductDto> products) {
		return new ExhibitionFindOneResponse(
			exhibition.getId(),
			exhibition.getTitle(),
			exhibition.getDescription(),
			exhibition.getVisibility(),
			exhibition.getStartAt(),
			exhibition.getEndAt(),
			products
		);
	}
}

File: src/main/kotlin/kr/co/lunatalk/domain/exhibition/dto/response/ExhibitionFindOneResponse.kt
package kr.co.lunatalk.domain.exhibition.dto.response

import io.swagger.v3.oas.annotations.media.Schema
import kr.co.lunatalk.domain.exhibition.domain.Exhibition
import kr.co.lunatalk.domain.exhibition.domain.ExhibitionVisibility
import kr.co.lunatalk.domain.exhibition.dto.ExhibitionProductDto
import java.time.LocalDateTime

data class ExhibitionFindOneResponse(
    @field:Schema(description = "기획전 ID")
    val exhibitionId: Long?,

    @field:Schema(description = "기획전 이름")
    val title: String,

    @field:Schema(description = "기획전 설명")
    val description: String?,

    @field:Schema(description = "기획전 노출 여부")
    val visibility: ExhibitionVisibility?,

    @field:Schema(description = "기획전 시작일")
    val startAt: LocalDateTime,

    @field:Schema(description = "기획전 종료일")
    val endAt: LocalDateTime?,

    @field:Schema(description = "연동된 상품들")
    val products: List<ExhibitionProductDto>
) {
    companion object {
        fun from(exhibition: Exhibition, products: List<ExhibitionProductDto>): ExhibitionFindOneResponse {
            return ExhibitionFindOneResponse(
                exhibitionId = exhibition.id,
                title = exhibition.title,
                description = exhibition.description,
                visibility = exhibition.visibility,
                startAt = exhibition.startAt,
                endAt = exhibition.endAt,
                products = products
            )
        }
    }
}
